### PR TITLE
[wip] minify js, css and pictures for better performances

### PR DESCRIPTION
**Pull request by ddurieux**
Impossible to minify + apped in only one file tiny_mce

Fix bugs with minify

Forget images to use with sprite

See performances here: https://siprossii.com/optimisation-de-glpi-0.85

### CODE REVIEW

**Comment by orthagh** [comment]
I can't merge in 0.85 only :/
Could you provide a version compatible with 0.90.

otherwise, i'll take only css/js minify part

**Comment by ddurieux** [comment]
Yep, I'm currently preparing for 0.90 but not yet ready

**Comment by wawax** [comment]
any news from this pr or can we close it ?

**Comment by ddurieux** [comment]
this is currently being finalized

**Comment by wawax** [comment]
ok great.
Please do it quickly before we stop accepting feature enhancement for 9.1

**Comment by remicollet** [comment]
Back from PHP Tour, talking a lot with build / deployment good practice.

I think this is not the way to go.

Working from git should use "normal" asset.
minimification and other asset optimization should be part of the "release" (aka build) process.

(and probably, most of this changes is mostly a workaround waiting for http/2)

**Comment by ddurieux** [comment]
You're right with optimization for the release, but it's that, prepare all to optimize for the release.

For the performances, see the link in first comment ;)

File: inc/html.class.php [hunk 1]
```diff
@@ -5372,5 +5364,85 @@ static function showCheckboxMatrix(array $columns, array $rows, array $options=a
...
+      if (isset($options['onclick'])) {
+          $onclick = "onclick='".$options['onclick']."'";
+      }
+
+      $style = '';
+      if (isset($options['style'])) {
+          $onclick = "style='".$options['style']."'";
+      }
+
+      $image = sprintf("<span class='block %s %s-%s %s' %s %s %s %s></span>",
```
Comment (remicollet) [inline inc/html.class.php]: I was expecting here a 

if (file_exists(main.png))
     use the sprite
else
     use the image

and main.png + man.css being generated during the release process.

**Comment by remicollet** [comment]
More: such file should be named "main-<sha1sum>.png" to be diff whand changed, and avoid cache issue.

**Comment by yllen** [comment]
@ddurieux and @remicollet you said the same thing.
This must be do for http/1 users.

@remicollet only said that the main.png must done only when you generated the release. Why? because is more simple for developer who want to add a new picture (not to regenerate .css, main.png...) just put the picture in pics directory.

Some task must be done during the release like the empty.sql that Moyo had generated from the upgrade to test it.

**Comment by remicollet** [comment]
> More: such file should be named "main-.png" to be different when changed,

Seems some part was missing (cleanup by github) in my previous messge.
I mean, should be called main-xxxx.png (where xxx is different when changed, e.g. sha1 of its content)

**Comment by orthagh** [comment]
I close this old PR.

For minification, we should enhance the release script

**Pull request closed without merging**